File: authorization.js
# THIS IS MY FUNCTION THAT WORKS IN GOOGLE APP SCRIPT. I NEED TO CONVERT IT TO PYTHON, BUT THE PYTHON ONE ISN'T WORKING.

function login() {
  
  var usrname = "MY EMAIL HERE";
  var pwd = "MY PASSWORD HERE";
  
  var URL = "https://api.pocketcasts.com/user/login";
  var origin = "https://play.pocketcasts.com";
  var payload = {email: usrname, password: pwd, scope: "webplayer"};
  
  var options = {
    "headers": {
      "Origin": origin
    },
    "method" : "POST",
    "payload": payload,
  }
  var token = JSON.parse(UrlFetchApp.fetch(URL, options).getContentText()).token;
  
  var tokenRange = SpreadsheetApp.getActiveSpreadsheet().getSheetByName('Settings').getRange("C5");
  tokenRange.setValues([[token]]);
  Logger.log(token);
  
  
}

File: authorization.py
import requests
import json

## this results in a 500 Internal Server Error and I don't know why.

username = "YOUR EMAIL HERE"
pwd = "YOUR PASSWORD HERE"

url = "https://api.pocketcasts.com/user/login"

origin = "https://play.pocketcasts.com"
payload = {"email": username, "password": pwd, "scope": "webplayer"}
headers = {"Origin": origin}

r = requests.post(url, headers=headers, params = payload)

print(r.status_code)
